src/Components/Search/SearchBar.tsx

```diff
@@ -1,104 +1,271 @@
+        node: { href },
+      },
     }
+  ) => {
+    window.location.href = href
   }
 
-  render() {
-    const { term } = this.state
+  renderPreview() {
+    const { entityID, entityType } = this.state
+    if (entityID && entityType) {
+      return <SearchPreview entityID={entityID} entityType={entityType} />
+    }
+  }
+
+  renderSuggestionsContainer = ({ containerProps, children, query }) => {
+    const { focused } = this.state
+
+    let firstItem = null
+    if (query) {
+      firstItem = <Box>Search "{query}"</Box>
+    } else if (focused) {
+      firstItem = <Box>{PLACEHOLDER}</Box>
+    }
 
     return (
-      <>
-        <Input
-          block
-          onInput={this.searchTextChanged.bind(this)}
-          onPaste={this.searchTextChanged.bind(this)}
-          onCut={this.searchTextChanged.bind(this)}
-          autoFocus
+      <Box {...containerProps}>
+        <Flex flexDirection={["column", "row"]}>
+          <Box width={["100%", "50%"]}>
+            <Flex flexDirection="column">
+              {firstItem}
+              {children}
+            </Flex>
+          </Box>
+          <Box width={["100%", "50%"]}>{this.renderPreview()}</Box>
+        </Flex>
+      </Box>
+    )
+  }
+
+  renderSuggestion = (
+    { node: { displayLabel, searchableType } },
+    { isHighlighted }
+  ) => {
+    return (
+      <Box style={{ backgroundColor: isHighlighted ? "#ddd" : "#fff" }}>
+        <SuggestionItem display={displayLabel} label={searchableType} />
+      </Box>
+    )
+  }
+
+  renderInputComponent = inputProps => (
+    <Box>
+      <Input style={{ width: "100%" }} {...inputProps} />
+    </Box>
+  )
+
+  renderAutosuggestComponent(xs: boolean) {
+    const { input } = this.state
+    const { viewer } = this.props
+    const edges = get(viewer, v => v.search.edges, [])
+
+    const inputProps = {
+      onChange: this.searchTextChanged,
+      onFocus: this.onFocus,
+      onBlur: this.onBlur,
+      placeholder: xs ? "" : PLACEHOLDER,
+      value: input,
+    }
+
+    return (
+      <AutosuggestContainer>
+        <Autosuggest
+          suggestions={edges}
+          onSuggestionsClearRequested={this.onSuggestionsClearRequested}
+          onSuggestionHighlighted={this.throttledOnSuggestionHighlighted}
+          onSuggestionsFetchRequested={this.throttledFetch}
+          getSuggestionValue={({ node: { displayLabel } }) => displayLabel}
+          renderSuggestion={this.renderSuggestion}
+          renderSuggestionsContainer={this.renderSuggestionsContainer}
+          inputProps={inputProps}
+          onSuggestionSelected={this.onSuggestionSelected}
+          renderInputComponent={this.renderInputComponent}
         />
-        <Spacer />
-        {this.resultsForTerm(term)}
+      </AutosuggestContainer>
+    )
+  }
+
+  render() {
+    return (
+      <>
+        <Media at="xs">{this.renderAutosuggestComponent(true)}</Media>
+        <Media greaterThan="xs">{this.renderAutosuggestComponent(false)}</Media>
```

Comment:
Seems like yr saying "don't render autoSuggest" -- better to always use objects rather than naked arguments; object props can also be rearranged and / or removed without having to refactor the receiver.

Reply:
I'll make these objects. Actually this is just to modify some props that go into `Autosuggest`, they're different for `xs` (placeholders).